common/src/main/java/tachyon/security/UserInformation.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.security;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.security.auth.login.AppConfigurationEntry;
+import javax.security.auth.login.Configuration;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import tachyon.Constants;
+
+/**
+ * This class wraps a JAAS Subject and provides methods to determine the user's name. It supports
```

Comment:
Sorry for making the confusion. This javadoc is targeted to the complete class, however this class is not complete now. Two `TODO` items are added in the latest patch for clarification.

You are right, `getOsPrincipalClassName` and `getOSLoginModuleName` are just utility functions. We will get user and its name in follow-up work here.